tools/python/boutiques/localExec.py

```diff
@@ -1047,6 +1047,68 @@ def _generateOutputFileNames(self):
                 outputFileName = os.path.abspath(outputFileName)
             self.out_dict[outputId] = outputFileName
 
+        # if 'conditional-path-template' in outputItem
+        # (key=conditions, value=path)
+        # Initialize file name with path template or existing value
+        for outputId in [x['id'] for x in self.outputs
+                         if 'conditional-path-template' in x]:
+
+            for boolObj in self.safeGet(outputId, 'conditional-path-template'):
+                # Surround expression chars with space
+                # Then split by space to isolate ids
+                boolExp = list(boolObj)[0]
+                splitExp = "".join(
+                    [c if c.isalnum() or c == "_" else
+                     " {0} ".format(c) for c in boolExp]).split(" ")
+                parsedExp = []
+                for word in [word.strip() for word in splitExp]:
```

Comment:
```suggestion
                for word in [word.strip() for word in splitExp if len(word) > 0]:
```

I'd do the above so that empty characters are thrown out here, before any processing.

Reply:
change will be included in next commit